pkg/epp/scheduling/types/types.go

```diff
@@ -62,8 +63,26 @@ type SchedulingContext struct {
 	Logger       logr.Logger
 	Req          *LLMRequest
 	PodsSnapshot []Pod
+	// PluginState can be used by plugins to store state during a scheduling cycle, to communicate
+	// between different extension points.
+	PluginState   map[PluginName]any
+	pluginStateMu *sync.RWMutex
```

Comment:
why not use sync.Map like in `CycleState` of Kube-Scheduler?
https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/cycle_state.go#L50

and then you don't need lock nor the Get/Set functions cause plugins can directly set or load values to/from the map concurrently (btw, atm plugins are running sequentially and not in parallel, but this could be changed easily)

Reply:
you lose type safety in sync.Map, I generally prefer a simple map with mutex unless we see performance bottlenecks. Given the number of plugins is small, I don't think there will be any performance issues any time soon.